src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const reducer = (reducerState, action) => {
+    switch (action.type) {
+      case 'addProperties':
+        return {
+          ...reducerState,
+          ...action.extraData,
+        };
+      case 'removeProperties':
+        const updatedState = action.keysToRemove.reduce((object, key) => {
+          const { [key]: _, ...p } = object;
+
+          return p;
+        }, reducerState);
+
+        return updatedState;
+      case 'clear':
+        return {};
```

Comment:
it's better to remove all key in existing object instead of creating a new one
with new one you will lose link to existing object